tests/ha/single_pod_restart/test_data_server_pod_restart.py

```diff
@@ -518,3 +524,407 @@ def test_io_data_server_pod_restart(self):
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 6.1: Performed READs/Verify on data written in healthy cluster in step-1")
+
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("STEP 6.2: Perform READs/Verify on data written in degraded mode when "
+                        "data pod down in step-4")
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix_deg,
+                                                        skipwrite=True, skipcleanup=True,
+                                                        setup_s3bench=False, nsamples=5, nclients=5)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 6.2: Performed READs/Verify on data written in degraded mode when "
+                        "data pod down in step-4")
+
+        LOGGER.info("STEP 7: Perform write when data and server pod is down")
+        if CMN_CFG["dtm0_disabled"]:
+            self.test_prefix = 'test-34076-deg-write'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix, skipread=True,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 7: Performed write when data and server pod is down")
+
+        LOGGER.info("Step 8: Start READs and verify DI on the written data when pods down in "
+                    "step-7 in background")
+        output_rd = Queue()
+        event = threading.Event()
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': self.test_prefix,
+                'nclients': 5, 'nsamples': 5, 'skipwrite': True, 'skipcleanup': True,
+                'output': output_rd, 'setup_s3bench': False}
+        thread_rd = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        thread_rd.daemon = True  # Daemonize thread
+        thread_rd.start()
+        LOGGER.info("Step 8: Successfully started READs and verify on the written data when pods "
+                    "down in step-7 in background")
+        LOGGER.info("Waiting for %s seconds", HA_CFG["common_params"]["30sec_delay"])
+        time.sleep(HA_CFG["common_params"]["30sec_delay"])
+
+        LOGGER.info("Step 9: Restore data and server pod and check cluster status.")
```

Comment:
for this test didn't we decide that we will have data pod restored and then run IOs only during server pod restart?